src/client/Microsoft.Identity.Client/ManagedIdentity/ManagedIdentityRequest.cs

```diff
@@ -13,9 +13,9 @@ namespace Microsoft.Identity.Client.ManagedIdentity
 {
     internal class ManagedIdentityRequest
     {
-        public Uri Endpoint { get; private set; }
+        public Lazy<Uri> Endpoint => new Lazy<Uri>(() => ComputeUri(this)) ;
```

Comment:
`Lazy` executes only on the first call. So if `Endpoint` is called, then some other query params are added, then `Endpoint is called again - it will return the original. This works correctly with the current code, but can introduce bugs down the line.

Reply:
The `ComputeUri` does not seem to be that expensive / used too much. If there are perf concerns, we should measure first and then optimize. 
Please remove the Lazy here, beucase the class is not immutable (all those collections are mutable because you can add / remove elements from them). Hence `ComputeUri` needs to always re-compute URI.

Reply:
Updated.